docs: update onboarding docs to include instructions for setting up intelliJ (#7052)
magma/magma: docs/readmes/contributing/contribute_onboarding.md

**Howto**

1. Follow the [prerequisites guide](https://magma.github.io/magma/docs/next/basics/prerequisites) and install all development tools, up to but not including the "Build/Deploy Tooling" section

> 2. (Optional) If you opt to use IntelliJ IDEA as your local IDE, follow the instructions in the "Set up IntelliJ" section below before you proceed

Reviewer: You can provide the direct link to this. Double check it (using the create_docusarus_website.sh script), but should be something like
```suggestion
2. (Optional) If you opt to use IntelliJ IDEA as your local IDE, follow the instructions in the [Set up IntelliJ](#set-up-intellij) section below before you proceed
```

Author: Doubled checked & verified that it works!

4. Create a Go module for the project by going to `Files > Project Structure > Project Settings > Modules > Click on "+" sign in the toolbar > New Module > Next`. When you reach the new module creation page, enter the following information:
    - Module name: `magma`
    - Content root, module file location: full path to your local Magma clone, e.g. `/Users/your_username/magma`

After completing the above steps, restart your IDE and ensure the environment is properly set up

1. Open "Project" on the left toolbar, and display "Project Files". All the files in the root `magma` directory should be displayed *without* a yellow background. This indicates IntelliJ recognizes the files as part of the module.
2. At the top-right corner of your IDE, you should see a drop-down menu showing a list of run configurations for the Magma test suites, with a green triangular button that allows you to run the selected test. Alternatively, when you open your run configurations (`Run > Edit Configurations`), you should see something like the below

>    ![intellij_initial_run_configs](assets/intellij_initial_run_configs.png).

Reviewer: ```suggestion
![intellij_initial_run_configs](assets/intellij_initial_run_configs.png).
```

> You can now run all (Orchestrator) test suites in one click.

Reviewer: ```suggestion
You can now run all (Orchestrator) tests in one click.
```

**Howto**

1. Follow the [prerequisites guide](https://magma.github.io/magma/docs/next/basics/prerequisites) and install all development tools, up to but not including the "Build/Deploy Tooling" section

> 2. (Optional) If you opt to use IntelliJ IDEA as your local IDE, follow the
>    instructions in the "Setting up IntelliJ" section below before you proceed

Reviewer: ```suggestion
2. (Optional) If you opt to use IntelliJ IDEA as your local IDE, follow the instructions in the "Set up IntelliJ" section below before you proceed
```

> 2. (Optional) If you opt to use IntelliJ IDEA as your local IDE, follow the
>    instructions in the "Setting up IntelliJ" section below before you proceed

Reviewer: ```suggestion
2. (Optional) If you opt to use IntelliJ IDEA as your local IDE, follow the instructions in the "Set up IntelliJ" section below before you proceed
```

3. Run all Orc8r tests
    1. Via Docker build script: `cd ${MAGMA_ROOT}/orc8r/cloud/docker && ./build.py -t ; noti`
    2. [Via IntelliJ](https://magma.github.io/magma/docs/orc8r/dev_testing#testing-tips)
4. Follow the [quick start guide](https://magma.github.io/magma/docs/next/basics/quick_start_guide) to get an AGW and Orc8r instance running on your dev machine
5. Visit the local [Swagger UI](https://swagger.io/tools/swagger-ui/) view of our REST API (URL is in @hcgatewood's Google Chrome bookmarks) and [list the set of managed networks](https://localhost:9443/apidocs/v1/#/Networks/get_networks) -- there should be one named "test"
    - You will need to toggle a Google Chrome preference to [allow insecure localhost](https://superuser.com/questions/772762/how-can-i-disable-security-checks-for-localhost)

Note: remember to periodically call `docker system prune` to clear outdated Docker artifacts from your dev machine.

> ### Setting Up IntelliJ

Reviewer: ```suggestion
### Set up IntelliJ
```

> ### Setting Up IntelliJ

Reviewer: ```suggestion
### Set up IntelliJ
```

> Although the development process could carry out in any IDE of your choice,
> we provide a set of run configurations via IntelliJ that helps running the test
> cases in a faster and more interactive manner. See
> [our Orchestrator testing tips](https://magma.github.io/magma/docs/orc8r/dev_testing#testing-tips)
> for more details.

Reviewer: ```suggestion
We recommend using [IntelliJ IDEA](https://www.jetbrains.com/idea/) for general Magma development, or [Visual Studio Code](https://code.visualstudio.com/) for a free alternative.

For IntelliJ IDEA, we provide a set of run configurations that support rapidly testing Magma code. See [Testing Tips](https://magma.github.io/magma/docs/orc8r/dev_testing#testing-tips) for more details.
```

> To set up your local IntelliJ environment, follow the following steps:

Reviewer: ```suggestion
To set up your local IntelliJ environment, perform the following

```

> 1. After downloading Magma, open the directory in IntelliJ

Reviewer: ```suggestion
1. After cloning the Magma repo, open the directory in IntelliJ
```

> 2. Ensure that the Go Plugin has been installed by going to Preferences > Plugins >
>    Search for the plugin "Go" > make sure it's been installed

Reviewer: ```suggestion
2. Ensure the [Go plugin](https://plugins.jetbrains.com/plugin/9568-go) has been installed by going to `Preferences > Plugins > search for the plugin "Go"`
```

> 2. Ensure that the Go Plugin has been installed by going to Preferences > Plugins >
>    Search for the plugin "Go" > make sure it's been installed

Reviewer: ```suggestion
2. Ensure the [Go plugin](https://plugins.jetbrains.com/plugin/9568-go) has been installed by going to `Preferences > Plugins > search for the plugin "Go"`
```

> 3. Specify the location of the Go SDK by going to Preferences > Languages &
>    Frameworks > Go > GOROOT > Select location for Go SDK in the drop-down menu.
>    [More information](https://www.jetbrains.com/help/idea/quick-start-guide-goland.html#step-1-open-or-create-a-project)

Reviewer: ```suggestion
3. [Specify the location of the Go SDK](https://www.jetbrains.com/help/idea/quick-start-guide-goland.html#step-1-open-or-create-a-project) by going to `Preferences > Languages & Frameworks > Go > GOROOT` and selecting the relevant location
```

> 4. Create a Go module for the project by going to Files > Project Structure > Project
>    Settings > Modules > Click on "+" sign in the toolbar > New Module > Next.
>    When you reach the New Module creation page, enter the following information:
>     - Module name: magma
>     - Content root & Module file location: the same as the path to your local
>       magma directory. For example, if the path to the root of your magma directory
>       is `/Users/your_username/magma`, then you should fill in the same path here
> 
>    Once you're finished, click "Finished" to apply the changes.

Reviewer: ```suggestion
4. Create a Go module for the project by going to `Files > Project Structure > Project Settings > Modules > Click on "+" sign in the toolbar > New Module > Next`. When you reach the new module creation page, enter the following information:
    - Module name: `magma`
    - Content root, module file location: full path to your local Magma clone, e.g. `/Users/your_username/magma`
```

> 4. Create a Go module for the project by going to Files > Project Structure > Project
>    Settings > Modules > Click on "+" sign in the toolbar > New Module > Next.
>    When you reach the New Module creation page, enter the following information:
>     - Module name: magma
>     - Content root & Module file location: the same as the path to your local
>       magma directory. For example, if the path to the root of your magma directory
>       is `/Users/your_username/magma`, then you should fill in the same path here
> 
>    Once you're finished, click "Finished" to apply the changes.

Reviewer: ```suggestion
4. Create a Go module for the project by going to `Files > Project Structure > Project Settings > Modules > Click on "+" sign in the toolbar > New Module > Next`. When you reach the new module creation page, enter the following information:
    - Module name: `magma`
    - Content root, module file location: full path to your local Magma clone, e.g. `/Users/your_username/magma`
```

> After you're finished with the above steps, restart your IDE and make sure the
> environment is properly set up by checking the following:

Reviewer: ```suggestion
After completing the above steps, restart your IDE and ensure the environment is properly set up
```

> After you're finished with the above steps, restart your IDE and make sure the
> environment is properly set up by checking the following:

Reviewer: ```suggestion
After completing the above steps, restart your IDE and ensure the environment is properly set up
```

> 1. When you open "Project" on the left toolbar, and display "Project Files",
> you should see all the files in your magma directory displayed without a yellow
> background color. This shows IntelliJ is recognizing the directory as a GO project.
> 2. At the top-right corner of your window, you should see a drop-down menu showing
> a list of run configurations for the magma test suites, with a green triangular
> button that allows you to run the selected test. Alternatively, when you
> open your run/deploy configurations (Run > Edit Configurations), you should see
> something like this:
> ![intellij_initial_run_configs](assets/intellij_initial_run_configs.png).
> Once this is successfully set up, you could run all of our default test suites
> in one click.

Reviewer: ```suggestion
1. Open "Project" on the left toolbar, and display "Project Files". All the files in the root `magma` directory should be displayed *without* a yellow background. This indicates IntelliJ recognizes the files as part of the module.
2. At the top-right corner of your IDE, you should see a drop-down menu showing a list of run configurations for the Magma test suites, with a green triangular button that allows you to run the selected test. Alternatively, when you open your run configurations (`Run > Edit Configurations`), you should see something like the below

![intellij_initial_run_configs](assets/intellij_initial_run_configs.png).

You can now run all (Orchestrator) test suites in one click.

NOTE: a minority of tests require a running Postgres instance, and will otherwise fail with connection errors. You can use `orc8r/cloud/docker/run.py` to spin up the required test DB.
```
